src/ApiCompat/Microsoft.DotNet.ApiCompatibility/Logging/Suppression.cs

```diff
@@ -57,9 +57,17 @@ private Suppression()
         /// the property won't be serialized to keep the baseline file minimal. The method's name is important as
         /// XmlSerializer will look for methods called ShouldSerializeX to determine if properties should be serialized.
         /// </summary>
-        /// <returns>Returns true if IsBaselineSuppression should be serialized</returns>
+        /// <returns>Returns true when IsBaselineSuppression should be serialized.</returns>
         public bool ShouldSerializeIsBaselineSuppression() => IsBaselineSuppression;
 
+        /// <summary>
+        /// Only serialize the DiagnosticId property when it is not empty. This can happen when a global suppression
```

Comment:
Does this mean that a suppression file with an empty diagnostic ID will lose that suppression when regenerating it?

Reply:
No. It means that a global suppression without a diagnostic id that is already stored in the suppression file does get a `<Diagnostic />` entry added when re-generating the suppression file.

Reply:
I see.  Got it, this keeps the generated file concise.  Makes sense.  Thanks for the explanation.